compiler/rustc_mir/src/const_eval/eval_queries.rs

```diff
@@ -105,7 +105,7 @@ pub(super) fn mk_eval_cx<'mir, 'tcx>(
 /// type system.
 pub(super) fn op_to_const<'tcx>(
     ecx: &CompileTimeEvalContext<'_, 'tcx>,
-    op: OpTy<'tcx>,
+    op: &OpTy<'tcx>,
```

Comment:
This kind of changes should already happen at ABI level.

Reply:
Couldn't it still make a difference on the caller side, which might have to worry about the callee mutating that memory when things are implicitly pass-by-ref?

Reply:
Yeah, I think that is the case.

Reply:
The `OpTy` is classified as a memory (see commit with size assertions) and is passed on the stack. If a function is merely forwarding an `OpTy` argument to another, it will have to copy it into a stack slot. What could be improved there?

Reply:
Ah, you are right about rust ABI. We are passing those indirectly without byval, but that means memcpyopt cannot eliminate the copy.